profile/src/main/java/rdx/works/profile/domain/GetFactorSourcesWithAccountsUseCase.kt

```diff
@@ -30,14 +31,16 @@ class GetFactorSourcesWithAccountsUseCase @Inject constructor(
                     val babylonPersonas = allPersonasOnNetwork.filter {
                         it.factorSourceId == deviceFactorSource.id && it.usesCurve25519
                     }
-                    result.add(
-                        DeviceFactorSourceData(
-                            deviceFactorSource = deviceFactorSource,
-                            accounts = babylonAccounts,
-                            isBabylon = true,
-                            personas = babylonPersonas
+                    if (hasBabylonSeedPhraseLength) {
```

Comment:
Maybe you could use directly `if (deviceFactorSource.hasBabylonSeedPhraseLength)`